evap/contributor/tests/test_views.py

```diff
@@ -179,6 +228,30 @@ def test_contributor_evaluation_edit(self):
         response = form.submit(expect_errors=True)
         self.assertEqual(response.status_code, 403)
 
+    @patch.object(webtest.Form, 'submit_fields', submit_fields_without_disabled)
+    def test_single_locked_questionnaire(self):
+        locked_questionnaire = baker.make(
+            Questionnaire,
+            type=Questionnaire.Type.TOP,
+            is_locked=True,
+            visibility=Questionnaire.Visibility.EDITORS,
+        )
+        responsible = UserProfile.objects.get(email='responsible@institution.example.com')
+        evaluation = baker.make(
+            Evaluation,
+            course=baker.make(Course, responsibles=[responsible]),
+            state='prepared',
+        )
+        evaluation.general_contribution.questionnaires.set([locked_questionnaire])
+
+        page = self.app.get(f'/contributor/evaluation/{evaluation.pk}/edit', user=responsible, status=200)
+        form = page.forms['evaluation-form']
+        response = form.submit(name='operation', value='save')
+        # TODO: Somehow assert that the form was submitted successfully.
+        # follow works in this case, but does not quite hit the point
+        response.follow()
```

Comment:
I added my regression test which currently has two flaws:

1. The assertion is a bit weird, but maybe you can come up with something nicer.
2. As webtest does wrongly submit disabled fields, I had to add a helper method.

(Also: sorry for breaking your build)

Reply:
i added an assert for the success message after saving as we usually do